framework/src/main/kotlin/com/cultureamp/eventsourcing/PostgresDatabaseEventStore.kt

```diff
@@ -1,28 +1,31 @@
-package eventsourcing
+package com.cultureamp.eventsourcing
 
 import com.fasterxml.jackson.databind.ObjectMapper
 import com.fasterxml.jackson.databind.SerializationFeature.WRITE_DATES_AS_TIMESTAMPS
 import com.fasterxml.jackson.datatype.joda.JodaModule
 import com.fasterxml.jackson.module.kotlin.registerKotlinModule
-import database.jsonb
 import org.jetbrains.exposed.exceptions.ExposedSQLException
 import org.jetbrains.exposed.sql.*
 import org.jetbrains.exposed.sql.transactions.transaction
 import org.joda.time.DateTime
 import java.util.*
 
-class DatabaseEventStore private constructor(private val db: Database) : EventStore {
+class PostgresDatabaseEventStore private constructor(private val db: Database) :
+    EventStore {
     companion object {
-        fun create(db: Database): DatabaseEventStore {
-            transaction(db) {
-                // TODO don't do this if pointing directly to Murmur DB or potentially introduce separate migrations
-                SchemaUtils.create(Events)
-            }
-            return DatabaseEventStore(db)
+        fun create(db: Database): PostgresDatabaseEventStore {
+            return PostgresDatabaseEventStore(db)
         }
     }
 
-    override lateinit var listeners: List<EventListener>
+    override val listeners: MutableList<EventListener> = mutableListOf()
+
+    override fun setup() {
+        transaction(db) {
+            // TODO don't do this if pointing directly to Murmur DB or potentially introduce separate migrations
+            SchemaUtils.createMissingTablesAndColumns(Events)
```

Comment:
I just noticed this - I think it's safer to use `SchemaUtils.create` because that will only create the table if it doesn't already exist. This will mutate existing tables in potentially unpredictable ways, and it may be better to introduce support for migrations if we want to do mutations. For the events table however I think we probably never want to mutate it once it's in place (I do have one change I need to make to the indices though before you go live)

Reply:
Gonna fix this one